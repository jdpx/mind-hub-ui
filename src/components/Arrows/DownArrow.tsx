import React from 'react'

import downArrow from '../../assets/down-arrow.svg'

interface Props {
    height?: number
    testid?: string
}

export default function DownArrow({ height = 30, testid = 'down-arrow' }: Props) {
    return <img src={downArrow} alt="Down Arrow" height={height} data-testid={testid} />
}
