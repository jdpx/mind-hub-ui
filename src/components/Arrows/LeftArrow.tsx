import React from 'react'

import leftArrow from '../../assets/left-arrow.svg'

interface Props {
    height?: number
    testid?: string
}

export default function LeftArrow({ height = 30, testid = 'left-arrow' }: Props) {
    return <img src={leftArrow} alt="Left Arrow" height={height} data-testid={testid} />
}
