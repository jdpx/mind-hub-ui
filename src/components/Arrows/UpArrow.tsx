import React from 'react'

import upArrow from '../../assets/up-arrow.svg'

interface Props {
    height?: number
    testid?: string
}

export default function UpArrow({ height = 30, testid = 'up-arrow' }: Props) {
    return <img src={upArrow} alt="Up Arrow" height={height} data-testid={testid} />
}
