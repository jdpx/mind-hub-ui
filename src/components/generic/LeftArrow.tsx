import React from 'react'

import leftArrow from '../../assets/left-arrow.svg'

interface Props {
    height?: number
    testid?: string
}

export default function LeftArrow({ height = 30, testid }: Props) {
    return <img src={leftArrow} alt="Left Arrow" height={height} data-testid={testid} />
}
