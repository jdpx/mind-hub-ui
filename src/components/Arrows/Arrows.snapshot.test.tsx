import React from 'react'
import { render } from '@testing-library/react'
import DownArrow from './DownArrow'
import UpArrow from './UpArrow'
import LeftArrow from './LeftArrow'

describe('Arrows Snapshots', () => {
    describe('DownArrow', () => {
        it('renders down arrow', () => {
            const { asFragment } = render(<DownArrow />)
            expect(asFragment()).toMatchSnapshot()
        })
    })

    describe('UpArrow', () => {
        it('renders up arrow', () => {
            const { asFragment } = render(<UpArrow />)
            expect(asFragment()).toMatchSnapshot()
        })
    })

    describe('LeftArrow', () => {
        it('renders left arrow', () => {
            const { asFragment } = render(<LeftArrow />)
            expect(asFragment()).toMatchSnapshot()
        })
    })
})
