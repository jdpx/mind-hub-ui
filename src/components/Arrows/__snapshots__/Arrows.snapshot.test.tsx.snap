// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrows Snapshots DownArrow renders down arrow 1`] = `
<DocumentFragment>
  <img
    alt="Down Arrow"
    data-testid="down-arrow"
    height="30"
    src="down-arrow.svg"
  />
</DocumentFragment>
`;

exports[`Arrows Snapshots LeftArrow renders left arrow 1`] = `
<DocumentFragment>
  <img
    alt="Left Arrow"
    data-testid="left-arrow"
    height="30"
    src="left-arrow.svg"
  />
</DocumentFragment>
`;

exports[`Arrows Snapshots UpArrow renders up arrow 1`] = `
<DocumentFragment>
  <img
    alt="Up Arrow"
    data-testid="up-arrow"
    height="30"
    src="up-arrow.svg"
  />
</DocumentFragment>
`;
